src/recon_test/test_blocks_on_cylindrical_projectors.cxx

```diff
@@ -871,6 +873,90 @@ BlocksTests::run_intersection_with_cylinder_test()
+  auto scanner_sptr = std::make_shared<Scanner>(Scanner::SAFIRDualRingPrototype);
+  {
+    scanner_sptr->set_average_depth_of_interaction(5);
+    scanner_sptr->set_num_axial_crystals_per_block(1);
+    scanner_sptr->set_axial_block_spacing(scanner_sptr->get_axial_crystal_spacing()
+                                          * scanner_sptr->get_num_axial_crystals_per_block());
+    scanner_sptr->set_transaxial_block_spacing(scanner_sptr->get_transaxial_crystal_spacing()
+                                               * scanner_sptr->get_num_transaxial_crystals_per_block());
+    scanner_sptr->set_num_axial_blocks_per_bucket(2);
+    scanner_sptr->set_num_rings(2 * 5);  // More than 1 bucket
+  }
+  auto segments = scanner_sptr->get_num_rings() - 1;
+  VectorWithOffset<int> num_axial_pos_per_segment(-segments, segments);
+  VectorWithOffset<int> min_ring_diff_v(-segments, segments);
+  VectorWithOffset<int> max_ring_diff_v(-segments, segments);
+  for (int i = -segments; i <= segments; i++)
+    {
+      min_ring_diff_v[i] = i;
+      max_ring_diff_v[i] = i;
+      num_axial_pos_per_segment[i] = scanner_sptr->get_num_rings() - abs(i);
+    }
+
+  scanner_sptr->set_scanner_geometry("BlocksOnCylindrical");
+  scanner_sptr->set_up();
+  auto projdata_info_sptr
+      = std::make_shared<ProjDataInfoBlocksOnCylindricalNoArcCorr>(scanner_sptr,
+                                                                   num_axial_pos_per_segment,
+                                                                   min_ring_diff_v,
+                                                                   max_ring_diff_v,
+                                                                   scanner_sptr->get_max_num_views(),
+                                                                   scanner_sptr->get_max_num_non_arccorrected_bins());
+  //  scanner_sptr->set_scanner_geometry("Cylindrical");
+  //  scanner_sptr->set_up();
+  //  auto projdata_info_sptr = std::make_shared<ProjDataInfoCylindricalNoArcCorr>(scanner_sptr,
+  //                                                                               num_axial_pos_per_segment,
+  //                                                                               min_ring_diff_v,
+  //                                                                               max_ring_diff_v,
+  //                                                                               scanner_sptr->get_max_num_views(),
+  //                                                                               scanner_sptr->get_max_num_non_arccorrected_bins());
+
+  auto projdata_sptr = std::make_shared<ProjDataInMemory>(std::make_shared<ExamInfo>(ImagingModality::PT), projdata_info_sptr);
+
+  auto origin = CartesianCoordinate3D<float>(0, 0, 0);
+  auto volume_dimensions = CartesianCoordinate3D<int>(-1, -1, -1);
+  auto volume = std::make_shared<VoxelsOnCartesianGrid<float>>(*projdata_info_sptr, 1, origin, volume_dimensions);
```

Comment:
I'd prefer to pass the exam_info along, to avoid warnings and future compatibility, so create an `exam_info_sptr` above and use it for both `proj_data_sptr` and `volume_sptr`